### PR TITLE
[User] should have email validation

### PR DESCRIPTION
**Pull request by RuslanHamidullin**
Fix of https://app.getsentry.com/bit-zesty-client-apps/qae/group/65975548/

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -17,7 +17,8 @@ def after_initialize
 
   validates :agreed_with_privacy_policy, acceptance: { allow_nil: false, accept: '1' }, on: :create
 
-  validates :role, :account, presence: true
+  validates :role, :account, :email, presence: true
+  validates :email, email: true
```
Comment (jaleszek) [inline app/models/user.rb]: `validatable` has not added the email format validation ?

Comment (RuslanHamidullin) [inline app/models/user.rb]: Looks like not, a we have:
```
devise :database_authenticatable, :registerable,
         :recoverable, :rememberable, :trackable, :validatable, :confirmable,
         :zxcvbnable
```

Comment (jaleszek) [inline app/models/user.rb]: checked on master:

```ActiveRecord::RecordInvalid: Validation failed: Email is invalid```

Comment (RuslanHamidullin) [inline app/models/user.rb]: but wouldn't raise with:
```
james@engagepr,co.uk
```

Comment (jaleszek) [inline app/models/user.rb]: Validation failed: Email This field cannot be blank

Comment (RuslanHamidullin) [inline app/models/user.rb]: Try:
```
james@engagepr,co.uk
```

Comment (matthewford) [inline app/models/user.rb]: What is the email_regexp in the devise config (that's what :validatable uses)

Comment (matthewford) [inline app/models/user.rb]: https://github.com/plataformatec/devise/blob/master/lib/devise/models/validatable.rb#L29-L31

Comment (RuslanHamidullin) [inline app/models/user.rb]: Looks like default is:
```
  # Regex to use to validate the email address
  # config.email_regexp = /\A([\w\.%\+\-]+)@([\w\-]+\.)+([\w]{2,})\z/i
```

Comment (RuslanHamidullin) [inline app/models/user.rb]: Gonna compare with email validator's regexp

Comment (RuslanHamidullin) [inline app/models/user.rb]: Email validator uses:
```
  def self.regexp(options = {})
    options = default_options.merge(options)

    name_validation = options[:strict_mode] ? "-\\p{L}\\d+._" : "^@\\s"

    /\A\s*([#{name_validation}]{1,64})@((?:[-\p{L}\d]+\.)+\p{L}{2,})\s*\z/i
  end
```
https://github.com/balexand/email_validator/blob/master/lib/email_validator.rb#L6

**Pull request closed without merging**